artifactory/commands/transferfiles/delayedartifactshandler.go

```diff
@@ -0,0 +1,209 @@
+package transferfiles
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/content"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+const (
+	// Max delayedArtifacts that will be written in a file
+	maxDelayedArtifactsInFile = 50000
+)
+
+// TransferDelayedArtifactsMng managing multi threads writing delayed deployment artifacts in order to keep the order of deployment.
+type TransferDelayedArtifactsMng struct {
+	// All go routines will write delayedArtifacts to this channel
+	delayedArtifactsChannel chan FileRepresentation
+
+	// Files containing delayed artifacts to upload later on.
+	filesToConsume []string
+	deployedWriter delayedArtifactWriter
+}
+
+type delayedArtifactWriter struct {
+	writer               *content.ContentWriter
+	delayedArtifactCount int
+	// In case we have multiple delayedArtifacts files - we index them
+	fileIndex int
+}
+
+func (mng *TransferDelayedArtifactsMng) initDelayedArtifactWriter() (err error) {
+	// Init the content writer which responsible for writing delayed artifacts - That means delayed artifacts that we should upload later, in a certain order.
+	writer, err := mng.newContentWriter()
```

Comment:
The fact that we create the writer here means that we will close it in a different function. This is not advisable, because it makes it hard to track where exactly it will be closed.
The right approach should be -
1. Create the writer
2. Close it inside a defer 

This may mean that would need to create and close it somewhere else. Maybe the "start" function could be a good place.